src/components/Shifts/AddShiftForm.tsx

```diff
@@ -81,34 +101,54 @@ export const AddShiftForm = ({port, terminal, interval, shiftForms, confirmHandl
               <Box key={shift.id}
                    sx={{mb: 2, p: 2, border: '1px solid #ccc', width: '300px', backgroundColor: '#E6E9F1'}}>
                 <Typography variant="h2" sx={{paddingBottom: '10px', fontSize: '24px'}}>Shift #{shift.id}</Typography>
+                {error && (shift.name === '') && (
+                  <Typography color="error" variant="body2">Please add shift name</Typography>
+                )}
+                {error && (shift.startTime === 'Select start time' || shift.startTime === '') && (
+                  <Typography color="error" variant="body2">Please select start time</Typography>
+                )}
+                {error && (shift.endTime === 'Select end time' || shift.endTime === '') && (
+                  <Typography color="error" variant="body2">Please select end time</Typography>
+                )}
+                {error && shift.startTime === shift.endTime && (
+                  <Typography color="error" variant="body2">Start time and end time cannot be the same</Typography>
+                )}
+                {shift.endTime < shift.startTime && (
+                  <Typography color="warning" variant="body2">This is a midnight shift spanning to the next day</Typography>
+                )}
                 <Grid container spacing={2}>
                   <Grid item xs={12}>
                     <Typography sx={{fontSize: '16px', fontWeight: 'bold'}}>Name of shift</Typography>
                     <TextField
                       fullWidth
                       value={shift.name}
+                      placeholder="Enter the shift name"
+                      autoFocus
                       onChange={(e) => handleChange(shift.id, 'name', e.target.value)}
-                      error={shift.name === ''}
-                      helperText={shift.name === '' ? 'Name is required' : ''}
+                      inputProps={{ 'data-cy': 'shift-name-input' }}
                     />
                   </Grid>
                   <Grid item xs={12}>
                     <Typography sx={{fontSize: '16px', fontWeight: 'bold'}}>Start time</Typography>
                     <Box>
                       <Select
                         variant="outlined"
-                        value={shift.startTime}
+                        value={timeOptions(interval).includes(shift.startTime) ? shift.startTime : "Select start time"}
```

Comment:
You're calling timeOptions in 3 places with in the form so would be good to store them in a const above the form to avoid the recalulations